src/main/java/bsh/Types.java

```diff
@@ -556,7 +556,7 @@ If fromValue is (or would be) Primitive.NULL then fromType should be null.
     public static Object castObject( Class<?> toType, Class<?> fromType, Object fromValue,
             int operation, boolean checkOnly ) throws UtilEvalError {
         // assignment to loose type, void type, or exactly same type
-        if ( toType == null || arrayElementType(toType) == arrayElementType(fromType) )
+        if ( toType == null || toType == fromType )
```

Comment:
However this condition does satisfy the comments, for what that's worth.

Reply:
Comment changed